include/envoy/http/filter.h

```diff
@@ -213,6 +213,15 @@ class StreamFilterCallbacks {
    */
   virtual Router::RouteConstSharedPtr route(const Router::RouteCallback& cb) PURE;
 
+  /**
+   * Sets the cached route for the current request to the passed-in RouteConstSharedPtr parameter.
+   *
+   * NOTE: Similar to route(const Router::RouteCallback& cb), this route that is set will be
+   * overridden by clearRouteCache() in subsequent filters. Usage is intended for filters at the end
+   * of the filter chain.
+   */
+  virtual void setRoute(const Router::RouteConstSharedPtr route) PURE;
```

Comment:
nit: don't use `const` for value parameters, just do `Router::RouteConstSharedPtr`

Reply:
Thanks for raising. This comment also triggered me to consider if we wanted this to be a reference parameter `const Router::RouteConstSharedPtr&` instead. I saw existing usages ([http/utility](https://github.com/envoyproxy/envoy/blob/main/source/common/http/utility.cc#L916-L943) and [ext_authz filter](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/ext_authz/ext_authz.cc#L378-L392), among others) of RouteConstSharedPtr being a reference param, and conducting the same `route->routeEntry()` calls.

After giving it some thought and consulting https://docs.microsoft.com/en-us/cpp/cpp/how-to-create-and-use-shared-ptr-instances?view=msvc-160#example-5, I think its best to keep it as is, passing the shared_ptr by value. Since we want the callee `setRoute` to be an owner, and we're not too concerned with the copy operation performance. Do you agree?